



<!DOCTYPE html>
<html lang="en" class=" is-copy-enabled is-u2f-enabled">
  <head prefix="og: http://ogp.me/ns# fb: http://ogp.me/ns/fb# object: http://ogp.me/ns/object# article: http://ogp.me/ns/article# profile: http://ogp.me/ns/profile#">
    <meta charset='utf-8'>
    

    <link crossorigin="anonymous" href="https://assets-cdn.github.com/assets/frameworks-31e369ccd2b23a1eccde83f03ef36eafff1f7b9025f6042ccac33a7915753de2.css" integrity="sha256-MeNpzNKyOh7M3oPwPvNur/8fe5Al9gQsysM6eRV1PeI=" media="all" rel="stylesheet" />
    <link crossorigin="anonymous" href="https://assets-cdn.github.com/assets/github-6a628757f656425789b747aa37ffe446cad8faaa6e1217ddc024f3969bddbca7.css" integrity="sha256-amKHV/ZWQleJt0eqN//kRsrY+qpuEhfdwCTzlpvdvKc=" media="all" rel="stylesheet" />
    
    
    <link crossorigin="anonymous" href="https://assets-cdn.github.com/assets/site-48a26f2c4b3f32dd9656723f5fdd558d42f6bcc5d13ba0a45ae025eabc9cd8b3.css" integrity="sha256-SKJvLEs/Mt2WVnI/X91VjUL2vMXRO6CkWuAl6ryc2LM=" media="all" rel="stylesheet" />
    

    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta http-equiv="Content-Language" content="en">
    <meta name="viewport" content="width=device-width">
    
    <title>Biostatistics/Gene analysis.Rmd at master · WPierrick/Biostatistics · GitHub</title>
    <link rel="search" type="application/opensearchdescription+xml" href="/opensearch.xml" title="GitHub">
    <link rel="fluid-icon" href="https://github.com/fluidicon.png" title="GitHub">
    <link rel="apple-touch-icon" href="/apple-touch-icon.png">
    <link rel="apple-touch-icon" sizes="57x57" href="/apple-touch-icon-57x57.png">
    <link rel="apple-touch-icon" sizes="60x60" href="/apple-touch-icon-60x60.png">
    <link rel="apple-touch-icon" sizes="72x72" href="/apple-touch-icon-72x72.png">
    <link rel="apple-touch-icon" sizes="76x76" href="/apple-touch-icon-76x76.png">
    <link rel="apple-touch-icon" sizes="114x114" href="/apple-touch-icon-114x114.png">
    <link rel="apple-touch-icon" sizes="120x120" href="/apple-touch-icon-120x120.png">
    <link rel="apple-touch-icon" sizes="144x144" href="/apple-touch-icon-144x144.png">
    <link rel="apple-touch-icon" sizes="152x152" href="/apple-touch-icon-152x152.png">
    <link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon-180x180.png">
    <meta property="fb:app_id" content="1401488693436528">

      <meta content="https://avatars1.githubusercontent.com/u/12656811?v=3&amp;s=400" name="twitter:image:src" /><meta content="@github" name="twitter:site" /><meta content="summary" name="twitter:card" /><meta content="WPierrick/Biostatistics" name="twitter:title" /><meta content="Biostatistics - Biostats using R, mostly linear regressions and data plots." name="twitter:description" />
      <meta content="https://avatars1.githubusercontent.com/u/12656811?v=3&amp;s=400" property="og:image" /><meta content="GitHub" property="og:site_name" /><meta content="object" property="og:type" /><meta content="WPierrick/Biostatistics" property="og:title" /><meta content="https://github.com/WPierrick/Biostatistics" property="og:url" /><meta content="Biostatistics - Biostats using R, mostly linear regressions and data plots." property="og:description" />
      <meta name="browser-stats-url" content="https://api.github.com/_private/browser/stats">
    <meta name="browser-errors-url" content="https://api.github.com/_private/browser/errors">
    <link rel="assets" href="https://assets-cdn.github.com/">
    
    <meta name="pjax-timeout" content="1000">
    
    <meta name="request-id" content="56EEDA65:23CD6:133C367E:58321C70" data-pjax-transient>

    <meta name="msapplication-TileImage" content="/windows-tile.png">
    <meta name="msapplication-TileColor" content="#ffffff">
    <meta name="selected-link" value="repo_source" data-pjax-transient>

    <meta name="google-site-verification" content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU">
<meta name="google-site-verification" content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA">
    <meta name="google-analytics" content="UA-3769691-2">

<meta content="collector.githubapp.com" name="octolytics-host" /><meta content="github" name="octolytics-app-id" /><meta content="56EEDA65:23CD6:133C367E:58321C70" name="octolytics-dimension-request_id" />
<meta content="/&lt;user-name&gt;/&lt;repo-name&gt;/blob/show" data-pjax-transient="true" name="analytics-location" />



  <meta class="js-ga-set" name="dimension1" content="Logged Out">



        <meta name="hostname" content="github.com">
    <meta name="user-login" content="">

        <meta name="expected-hostname" content="github.com">
      <meta name="js-proxy-site-detection-payload" content="MGIyNDJlMGMyMWViN2ViNzMzZTVhMGU5MGJkMTRkODVkNTBkZGY1ZDcyN2NmZGQ5MmU3ZTdmYjFhYmY2NmU0MXx7InJlbW90ZV9hZGRyZXNzIjoiODYuMjM4LjIxOC4xMDEiLCJyZXF1ZXN0X2lkIjoiNTZFRURBNjU6MjNDRDY6MTMzQzM2N0U6NTgzMjFDNzAiLCJ0aW1lc3RhbXAiOjE0Nzk2NzkwOTMsImhvc3QiOiJnaXRodWIuY29tIn0=">


      <link rel="mask-icon" href="https://assets-cdn.github.com/pinned-octocat.svg" color="#000000">
      <link rel="icon" type="image/x-icon" href="https://assets-cdn.github.com/favicon.ico">

    <meta name="html-safe-nonce" content="f7c25936f4f9d102b10eb2555a449d5baf2ee300">
    <meta content="b36fdba80b77d8b72489d7c9c52465557cb41472" name="form-nonce" />

    <meta http-equiv="x-pjax-version" content="bbcf531292370b32e4fed7e4a29fe988">
    

      
  <meta name="description" content="Biostatistics - Biostats using R, mostly linear regressions and data plots.">
  <meta name="go-import" content="github.com/WPierrick/Biostatistics git https://github.com/WPierrick/Biostatistics.git">

  <meta content="12656811" name="octolytics-dimension-user_id" /><meta content="WPierrick" name="octolytics-dimension-user_login" /><meta content="44260997" name="octolytics-dimension-repository_id" /><meta content="WPierrick/Biostatistics" name="octolytics-dimension-repository_nwo" /><meta content="true" name="octolytics-dimension-repository_public" /><meta content="false" name="octolytics-dimension-repository_is_fork" /><meta content="44260997" name="octolytics-dimension-repository_network_root_id" /><meta content="WPierrick/Biostatistics" name="octolytics-dimension-repository_network_root_nwo" />
  <link href="https://github.com/WPierrick/Biostatistics/commits/master.atom" rel="alternate" title="Recent Commits to Biostatistics:master" type="application/atom+xml">


      <link rel="canonical" href="https://github.com/WPierrick/Biostatistics/blob/master/Gene%20analysis.Rmd" data-pjax-transient>
  </head>


  <body class="logged-out  env-production windows vis-public page-blob">
    <div id="js-pjax-loader-bar" class="pjax-loader-bar"><div class="progress"></div></div>
    <a href="#start-of-content" tabindex="1" class="accessibility-aid js-skip-to-content">Skip to content</a>

    
    
    



          <header class="site-header js-details-container" role="banner">
  <div class="container-responsive">
    <a class="header-logo-invertocat" href="https://github.com/" aria-label="Homepage" data-ga-click="(Logged out) Header, go to homepage, icon:logo-wordmark">
      <svg aria-hidden="true" class="octicon octicon-mark-github" height="32" version="1.1" viewBox="0 0 16 16" width="32"><path fill-rule="evenodd" d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0 0 16 8c0-4.42-3.58-8-8-8z"/></svg>
    </a>

    <button class="btn-link float-right site-header-toggle js-details-target" type="button" aria-label="Toggle navigation">
      <svg aria-hidden="true" class="octicon octicon-three-bars" height="24" version="1.1" viewBox="0 0 12 16" width="18"><path fill-rule="evenodd" d="M11.41 9H.59C0 9 0 8.59 0 8c0-.59 0-1 .59-1H11.4c.59 0 .59.41.59 1 0 .59 0 1-.59 1h.01zm0-4H.59C0 5 0 4.59 0 4c0-.59 0-1 .59-1H11.4c.59 0 .59.41.59 1 0 .59 0 1-.59 1h.01zM.59 11H11.4c.59 0 .59.41.59 1 0 .59 0 1-.59 1H.59C0 13 0 12.59 0 12c0-.59 0-1 .59-1z"/></svg>
    </button>

    <div class="site-header-menu">
      <nav class="site-header-nav site-header-nav-main">
        <a href="/personal" class="js-selected-navigation-item nav-item nav-item-personal" data-ga-click="Header, click, Nav menu - item:personal" data-selected-links="/personal /personal">
          Personal
</a>        <a href="/open-source" class="js-selected-navigation-item nav-item nav-item-opensource" data-ga-click="Header, click, Nav menu - item:opensource" data-selected-links="/open-source /open-source">
          Open source
</a>        <a href="/business" class="js-selected-navigation-item nav-item nav-item-business" data-ga-click="Header, click, Nav menu - item:business" data-selected-links="/business /business/partners /business/features /business/customers /business">
          Business
</a>        <a href="/explore" class="js-selected-navigation-item nav-item nav-item-explore" data-ga-click="Header, click, Nav menu - item:explore" data-selected-links="/explore /trending /trending/developers /integrations /integrations/feature/code /integrations/feature/collaborate /integrations/feature/ship /showcases /explore">
          Explore
</a>      </nav>

      <div class="site-header-actions">
            <a class="btn btn-primary site-header-actions-btn" href="/join?source=header-repo" data-ga-click="(Logged out) Header, clicked Sign up, text:sign-up">Sign up</a>
          <a class="btn site-header-actions-btn mr-1" href="/login?return_to=%2FWPierrick%2FBiostatistics%2Fblob%2Fmaster%2FGene%2520analysis.Rmd" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in">Sign in</a>
      </div>

        <nav class="site-header-nav site-header-nav-secondary mr-md-3">
          <a class="nav-item" href="/pricing">Pricing</a>
          <a class="nav-item" href="/blog">Blog</a>
          <a class="nav-item" href="https://help.github.com">Support</a>
          <a class="nav-item header-search-link" href="https://github.com/search">Search GitHub</a>
              <div class="header-search scoped-search site-scoped-search js-site-search" role="search">
  <!-- '"` --><!-- </textarea></xmp> --></option></form><form accept-charset="UTF-8" action="/WPierrick/Biostatistics/search" class="js-site-search-form" data-scoped-search-url="/WPierrick/Biostatistics/search" data-unscoped-search-url="/search" method="get"><div style="margin:0;padding:0;display:inline"><input name="utf8" type="hidden" value="&#x2713;" /></div>
    <label class="form-control header-search-wrapper js-chromeless-input-container">
      <div class="header-search-scope">This repository</div>
      <input type="text"
        class="form-control header-search-input js-site-search-focus js-site-search-field is-clearable"
        data-hotkey="s"
        name="q"
        placeholder="Search"
        aria-label="Search this repository"
        data-unscoped-placeholder="Search GitHub"
        data-scoped-placeholder="Search"
        autocapitalize="off">
    </label>
</form></div>

        </nav>
    </div>
  </div>
</header>



    <div id="start-of-content" class="accessibility-aid"></div>

      <div id="js-flash-container">
</div>


    <div role="main">
        <div itemscope itemtype="http://schema.org/SoftwareSourceCode">
    <div id="js-repo-pjax-container" data-pjax-container>
      
<div class="pagehead repohead instapaper_ignore readability-menu experiment-repo-nav">
  <div class="container repohead-details-container">

    

<ul class="pagehead-actions">

  <li>
      <a href="/login?return_to=%2FWPierrick%2FBiostatistics"
    class="btn btn-sm btn-with-count tooltipped tooltipped-n"
    aria-label="You must be signed in to watch a repository" rel="nofollow">
    <svg aria-hidden="true" class="octicon octicon-eye" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M8.06 2C3 2 0 8 0 8s3 6 8.06 6C13 14 16 8 16 8s-3-6-7.94-6zM8 12c-2.2 0-4-1.78-4-4 0-2.2 1.8-4 4-4 2.22 0 4 1.8 4 4 0 2.22-1.78 4-4 4zm2-4c0 1.11-.89 2-2 2-1.11 0-2-.89-2-2 0-1.11.89-2 2-2 1.11 0 2 .89 2 2z"/></svg>
    Watch
  </a>
  <a class="social-count" href="/WPierrick/Biostatistics/watchers"
     aria-label="1 user is watching this repository">
    1
  </a>

  </li>

  <li>
      <a href="/login?return_to=%2FWPierrick%2FBiostatistics"
    class="btn btn-sm btn-with-count tooltipped tooltipped-n"
    aria-label="You must be signed in to star a repository" rel="nofollow">
    <svg aria-hidden="true" class="octicon octicon-star" height="16" version="1.1" viewBox="0 0 14 16" width="14"><path fill-rule="evenodd" d="M14 6l-4.9-.64L7 1 4.9 5.36 0 6l3.6 3.26L2.67 14 7 11.67 11.33 14l-.93-4.74z"/></svg>
    Star
  </a>

    <a class="social-count js-social-count" href="/WPierrick/Biostatistics/stargazers"
      aria-label="0 users starred this repository">
      0
    </a>

  </li>

  <li>
      <a href="/login?return_to=%2FWPierrick%2FBiostatistics"
        class="btn btn-sm btn-with-count tooltipped tooltipped-n"
        aria-label="You must be signed in to fork a repository" rel="nofollow">
        <svg aria-hidden="true" class="octicon octicon-repo-forked" height="16" version="1.1" viewBox="0 0 10 16" width="10"><path fill-rule="evenodd" d="M8 1a1.993 1.993 0 0 0-1 3.72V6L5 8 3 6V4.72A1.993 1.993 0 0 0 2 1a1.993 1.993 0 0 0-1 3.72V6.5l3 3v1.78A1.993 1.993 0 0 0 5 15a1.993 1.993 0 0 0 1-3.72V9.5l3-3V4.72A1.993 1.993 0 0 0 8 1zM2 4.2C1.34 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zm3 10c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zm3-10c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"/></svg>
        Fork
      </a>

    <a href="/WPierrick/Biostatistics/network" class="social-count"
       aria-label="0 users forked this repository">
      0
    </a>
  </li>
</ul>

    <h1 class="public ">
  <svg aria-hidden="true" class="octicon octicon-repo" height="16" version="1.1" viewBox="0 0 12 16" width="12"><path fill-rule="evenodd" d="M4 9H3V8h1v1zm0-3H3v1h1V6zm0-2H3v1h1V4zm0-2H3v1h1V2zm8-1v12c0 .55-.45 1-1 1H6v2l-1.5-1.5L3 16v-2H1c-.55 0-1-.45-1-1V1c0-.55.45-1 1-1h10c.55 0 1 .45 1 1zm-1 10H1v2h2v-1h3v1h5v-2zm0-10H2v9h9V1z"/></svg>
  <span class="author" itemprop="author"><a href="/WPierrick" class="url fn" rel="author">WPierrick</a></span><!--
--><span class="path-divider">/</span><!--
--><strong itemprop="name"><a href="/WPierrick/Biostatistics" data-pjax="#js-repo-pjax-container">Biostatistics</a></strong>

</h1>

  </div>
  <div class="container">
    
<nav class="reponav js-repo-nav js-sidenav-container-pjax"
     itemscope
     itemtype="http://schema.org/BreadcrumbList"
     role="navigation"
     data-pjax="#js-repo-pjax-container">

  <span itemscope itemtype="http://schema.org/ListItem" itemprop="itemListElement">
    <a href="/WPierrick/Biostatistics" aria-selected="true" class="js-selected-navigation-item selected reponav-item" data-hotkey="g c" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches /WPierrick/Biostatistics" itemprop="url">
      <svg aria-hidden="true" class="octicon octicon-code" height="16" version="1.1" viewBox="0 0 14 16" width="14"><path fill-rule="evenodd" d="M9.5 3L8 4.5 11.5 8 8 11.5 9.5 13 14 8 9.5 3zm-5 0L0 8l4.5 5L6 11.5 2.5 8 6 4.5 4.5 3z"/></svg>
      <span itemprop="name">Code</span>
      <meta itemprop="position" content="1">
</a>  </span>

    <span itemscope itemtype="http://schema.org/ListItem" itemprop="itemListElement">
      <a href="/WPierrick/Biostatistics/issues" class="js-selected-navigation-item reponav-item" data-hotkey="g i" data-selected-links="repo_issues repo_labels repo_milestones /WPierrick/Biostatistics/issues" itemprop="url">
        <svg aria-hidden="true" class="octicon octicon-issue-opened" height="16" version="1.1" viewBox="0 0 14 16" width="14"><path fill-rule="evenodd" d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"/></svg>
        <span itemprop="name">Issues</span>
        <span class="counter">0</span>
        <meta itemprop="position" content="2">
</a>    </span>

  <span itemscope itemtype="http://schema.org/ListItem" itemprop="itemListElement">
    <a href="/WPierrick/Biostatistics/pulls" class="js-selected-navigation-item reponav-item" data-hotkey="g p" data-selected-links="repo_pulls /WPierrick/Biostatistics/pulls" itemprop="url">
      <svg aria-hidden="true" class="octicon octicon-git-pull-request" height="16" version="1.1" viewBox="0 0 12 16" width="12"><path fill-rule="evenodd" d="M11 11.28V5c-.03-.78-.34-1.47-.94-2.06C9.46 2.35 8.78 2.03 8 2H7V0L4 3l3 3V4h1c.27.02.48.11.69.31.21.2.3.42.31.69v6.28A1.993 1.993 0 0 0 10 15a1.993 1.993 0 0 0 1-3.72zm-1 2.92c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zM4 3c0-1.11-.89-2-2-2a1.993 1.993 0 0 0-1 3.72v6.56A1.993 1.993 0 0 0 2 15a1.993 1.993 0 0 0 1-3.72V4.72c.59-.34 1-.98 1-1.72zm-.8 10c0 .66-.55 1.2-1.2 1.2-.65 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2zM2 4.2C1.34 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"/></svg>
      <span itemprop="name">Pull requests</span>
      <span class="counter">0</span>
      <meta itemprop="position" content="3">
</a>  </span>

  <a href="/WPierrick/Biostatistics/projects" class="js-selected-navigation-item reponav-item" data-selected-links="repo_projects new_repo_project repo_project /WPierrick/Biostatistics/projects">
    <svg aria-hidden="true" class="octicon octicon-project" height="16" version="1.1" viewBox="0 0 15 16" width="15"><path fill-rule="evenodd" d="M10 12h3V2h-3v10zm-4-2h3V2H6v8zm-4 4h3V2H2v12zm-1 1h13V1H1v14zM14 0H1a1 1 0 0 0-1 1v14a1 1 0 0 0 1 1h13a1 1 0 0 0 1-1V1a1 1 0 0 0-1-1z"/></svg>
    Projects
    <span class="counter">0</span>
</a>


  <a href="/WPierrick/Biostatistics/pulse" class="js-selected-navigation-item reponav-item" data-selected-links="pulse /WPierrick/Biostatistics/pulse">
    <svg aria-hidden="true" class="octicon octicon-pulse" height="16" version="1.1" viewBox="0 0 14 16" width="14"><path fill-rule="evenodd" d="M11.5 8L8.8 5.4 6.6 8.5 5.5 1.6 2.38 8H0v2h3.6l.9-1.8.9 5.4L9 8.5l1.6 1.5H14V8z"/></svg>
    Pulse
</a>
  <a href="/WPierrick/Biostatistics/graphs" class="js-selected-navigation-item reponav-item" data-selected-links="repo_graphs repo_contributors /WPierrick/Biostatistics/graphs">
    <svg aria-hidden="true" class="octicon octicon-graph" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M16 14v1H0V0h1v14h15zM5 13H3V8h2v5zm4 0H7V3h2v10zm4 0h-2V6h2v7z"/></svg>
    Graphs
</a>

</nav>

  </div>
</div>

<div class="container new-discussion-timeline experiment-repo-nav">
  <div class="repository-content">

    

<a href="/WPierrick/Biostatistics/blob/1a089658aa92ee1462dbcd8605b84cfd9932cf72/Gene%20analysis.Rmd" class="d-none js-permalink-shortcut" data-hotkey="y">Permalink</a>

<!-- blob contrib key: blob_contributors:v21:c858b900e13c50d7ffa124ec52244983 -->

<div class="file-navigation js-zeroclipboard-container">
  
<div class="select-menu branch-select-menu js-menu-container js-select-menu float-left">
  <button class="btn btn-sm select-menu-button js-menu-target css-truncate" data-hotkey="w"
    
    type="button" aria-label="Switch branches or tags" tabindex="0" aria-haspopup="true">
    <i>Branch:</i>
    <span class="js-select-button css-truncate-target">master</span>
  </button>

  <div class="select-menu-modal-holder js-menu-content js-navigation-container" data-pjax aria-hidden="true">

    <div class="select-menu-modal">
      <div class="select-menu-header">
        <svg aria-label="Close" class="octicon octicon-x js-menu-close" height="16" role="img" version="1.1" viewBox="0 0 12 16" width="12"><path fill-rule="evenodd" d="M7.48 8l3.75 3.75-1.48 1.48L6 9.48l-3.75 3.75-1.48-1.48L4.52 8 .77 4.25l1.48-1.48L6 6.52l3.75-3.75 1.48 1.48z"/></svg>
        <span class="select-menu-title">Switch branches/tags</span>
      </div>

      <div class="select-menu-filters">
        <div class="select-menu-text-filter">
          <input type="text" aria-label="Filter branches/tags" id="context-commitish-filter-field" class="form-control js-filterable-field js-navigation-enable" placeholder="Filter branches/tags">
        </div>
        <div class="select-menu-tabs">
          <ul>
            <li class="select-menu-tab">
              <a href="#" data-tab-filter="branches" data-filter-placeholder="Filter branches/tags" class="js-select-menu-tab" role="tab">Branches</a>
            </li>
            <li class="select-menu-tab">
              <a href="#" data-tab-filter="tags" data-filter-placeholder="Find a tag…" class="js-select-menu-tab" role="tab">Tags</a>
            </li>
          </ul>
        </div>
      </div>

      <div class="select-menu-list select-menu-tab-bucket js-select-menu-tab-bucket" data-tab-filter="branches" role="menu">

        <div data-filterable-for="context-commitish-filter-field" data-filterable-type="substring">


            <a class="select-menu-item js-navigation-item js-navigation-open selected"
               href="/WPierrick/Biostatistics/blob/master/Gene%20analysis.Rmd"
               data-name="master"
               data-skip-pjax="true"
               rel="nofollow">
              <svg aria-hidden="true" class="octicon octicon-check select-menu-item-icon" height="16" version="1.1" viewBox="0 0 12 16" width="12"><path fill-rule="evenodd" d="M12 5l-8 8-4-4 1.5-1.5L4 10l6.5-6.5z"/></svg>
              <span class="select-menu-item-text css-truncate-target js-select-menu-filter-text">
                master
              </span>
            </a>
        </div>

          <div class="select-menu-no-results">Nothing to show</div>
      </div>

      <div class="select-menu-list select-menu-tab-bucket js-select-menu-tab-bucket" data-tab-filter="tags">
        <div data-filterable-for="context-commitish-filter-field" data-filterable-type="substring">


        </div>

        <div class="select-menu-no-results">Nothing to show</div>
      </div>

    </div>
  </div>
</div>

  <div class="BtnGroup float-right">
    <a href="/WPierrick/Biostatistics/find/master"
          class="js-pjax-capture-input btn btn-sm BtnGroup-item"
          data-pjax
          data-hotkey="t">
      Find file
    </a>
    <button aria-label="Copy file path to clipboard" class="js-zeroclipboard btn btn-sm BtnGroup-item tooltipped tooltipped-s" data-copied-hint="Copied!" type="button">Copy path</button>
  </div>
  <div class="breadcrumb js-zeroclipboard-target">
    <span class="repo-root js-repo-root"><span class="js-path-segment"><a href="/WPierrick/Biostatistics"><span>Biostatistics</span></a></span></span><span class="separator">/</span><strong class="final-path">Gene analysis.Rmd</strong>
  </div>
</div>


  <div class="commit-tease">
      <span class="float-right">
        <a class="commit-tease-sha" href="/WPierrick/Biostatistics/commit/afdd99cac20653876fd43a7557fac3bc187ab68f" data-pjax>
          afdd99c
        </a>
        <relative-time datetime="2015-12-07T22:53:37Z">Dec 7, 2015</relative-time>
      </span>
      <div>
        <img alt="@WPierrick" class="avatar" height="20" src="https://avatars0.githubusercontent.com/u/12656811?v=3&amp;s=40" width="20" />
        <a href="/WPierrick" class="user-mention" rel="author">WPierrick</a>
          <a href="/WPierrick/Biostatistics/commit/afdd99cac20653876fd43a7557fac3bc187ab68f" class="message" data-pjax="true" title="Analysis on gene expression values">Analysis on gene expression values</a>
      </div>

    <div class="commit-tease-contributors">
      <button type="button" class="btn-link muted-link contributors-toggle" data-facebox="#blob_contributors_box">
        <strong>1</strong>
         contributor
      </button>
      
    </div>

    <div id="blob_contributors_box" style="display:none">
      <h2 class="facebox-header" data-facebox-id="facebox-header">Users who have contributed to this file</h2>
      <ul class="facebox-user-list" data-facebox-id="facebox-description">
          <li class="facebox-user-list-item">
            <img alt="@WPierrick" height="24" src="https://avatars2.githubusercontent.com/u/12656811?v=3&amp;s=48" width="24" />
            <a href="/WPierrick">WPierrick</a>
          </li>
      </ul>
    </div>
  </div>


<div class="file">
  <div class="file-header">
  <div class="file-actions">

    <div class="BtnGroup">
      <a href="/WPierrick/Biostatistics/raw/master/Gene%20analysis.Rmd" class="btn btn-sm BtnGroup-item" id="raw-url">Raw</a>
        <a href="/WPierrick/Biostatistics/blame/master/Gene%20analysis.Rmd" class="btn btn-sm js-update-url-with-hash BtnGroup-item">Blame</a>
      <a href="/WPierrick/Biostatistics/commits/master/Gene%20analysis.Rmd" class="btn btn-sm BtnGroup-item" rel="nofollow">History</a>
    </div>

        <a class="btn-octicon tooltipped tooltipped-nw"
           href="https://windows.github.com"
           aria-label="Open this file in GitHub Desktop"
           data-ga-click="Repository, open with desktop, type:windows">
            <svg aria-hidden="true" class="octicon octicon-device-desktop" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M15 2H1c-.55 0-1 .45-1 1v9c0 .55.45 1 1 1h5.34c-.25.61-.86 1.39-2.34 2h8c-1.48-.61-2.09-1.39-2.34-2H15c.55 0 1-.45 1-1V3c0-.55-.45-1-1-1zm0 9H1V3h14v8z"/></svg>
        </a>

        <button type="button" class="btn-octicon disabled tooltipped tooltipped-nw"
          aria-label="You must be signed in to make or propose changes">
          <svg aria-hidden="true" class="octicon octicon-pencil" height="16" version="1.1" viewBox="0 0 14 16" width="14"><path fill-rule="evenodd" d="M0 12v3h3l8-8-3-3-8 8zm3 2H1v-2h1v1h1v1zm10.3-9.3L12 6 9 3l1.3-1.3a.996.996 0 0 1 1.41 0l1.59 1.59c.39.39.39 1.02 0 1.41z"/></svg>
        </button>
        <button type="button" class="btn-octicon btn-octicon-danger disabled tooltipped tooltipped-nw"
          aria-label="You must be signed in to make or propose changes">
          <svg aria-hidden="true" class="octicon octicon-trashcan" height="16" version="1.1" viewBox="0 0 12 16" width="12"><path fill-rule="evenodd" d="M11 2H9c0-.55-.45-1-1-1H5c-.55 0-1 .45-1 1H2c-.55 0-1 .45-1 1v1c0 .55.45 1 1 1v9c0 .55.45 1 1 1h7c.55 0 1-.45 1-1V5c.55 0 1-.45 1-1V3c0-.55-.45-1-1-1zm-1 12H3V5h1v8h1V5h1v8h1V5h1v8h1V5h1v9zm1-10H2V3h9v1z"/></svg>
        </button>
  </div>

  <div class="file-info">
      303 lines (188 sloc)
      <span class="file-info-divider"></span>
    8.64 KB
  </div>
</div>

  
  <div id="readme" class="readme blob instapaper_body">
    <article class="markdown-body entry-content" itemprop="text"><table data-table-type="yaml-metadata">
  <thead>
  <tr>
  <th>title</th>

  <th>author</th>

  <th>date</th>

  <th>output</th>
  </tr>
  </thead>
  <tbody>
  <tr>
  <td><div>Genomics data</div></td>

  <td><div>WAINSCHTEIN Pierrick</div></td>

  <td><div>16 novembre 2015</div></td>

  <td><div>pdf_document</div></td>
  </tr>
  </tbody>
</table>

<h2><a id="user-content-chargement-des-données" class="anchor" href="#chargement-des-données" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Chargement des données</h2>

<p>Nous nous intéressons tout d'abord à la structure de nos données.
Nous pouvons voir que le fichier liver.rda contient un objet qui contient une liste de 4 éléments :</p>

<ul>
<li>Gènes (contient les niveaux d'expression de 3116 gènes)</li>
<li>Clinic (contient les données sur les sujets de test ainsi que Cholesterol.mg.dL en dernière colonne)</li>
<li>Treatment (des informations sur les traitements utilisés)</li>
<li>Gene.ID (des informations pour retrouver les différents gènes dans la base de donnée NCBI)</li>
</ul>

<div class="highlight highlight-source-r"><pre>load(<span class="pl-s"><span class="pl-pds">"</span>liver.rda<span class="pl-pds">"</span></span>)

library(<span class="pl-smi">GGally</span>)
library(<span class="pl-smi">ggplot2</span>)


<span class="pl-c"># head(liver.toxicity)</span>
str(<span class="pl-smi">liver.toxicity</span>)

<span class="pl-smi">cholesterol</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">liver.toxicity</span><span class="pl-k">$</span><span class="pl-smi">clinic</span>[,<span class="pl-s"><span class="pl-pds">"</span>Cholesterol.mg.dL.<span class="pl-pds">"</span></span>]

str(<span class="pl-smi">liver.toxicity</span><span class="pl-k">$</span><span class="pl-smi">treatment</span>)

<span class="pl-smi">gene</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">liver.toxicity</span><span class="pl-k">$</span><span class="pl-smi">gene</span>

</pre></div>

<p>\newpage</p>

<h2><a id="user-content-analyse-descriptive-des-données" class="anchor" href="#analyse-descriptive-des-données" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Analyse descriptive des données</h2>

<p>On observe que les données de cholestérol semblent centrées autour de la valeur 80.</p>

<div class="highlight highlight-source-r"><pre>par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>c(<span class="pl-c1">1</span>,<span class="pl-c1">2</span>))
hist(<span class="pl-smi">cholesterol</span>,<span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>cyan<span class="pl-pds">"</span></span>, <span class="pl-v">breaks</span> <span class="pl-k">=</span> <span class="pl-c1">10</span>, <span class="pl-v">main</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>histogramme<span class="pl-pds">"</span></span>,<span class="pl-v">xlab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>)
</pre></div>

<h1><a id="user-content-sélection-des-données" class="anchor" href="#sélection-des-données" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Sélection des données</h1>

<p>Nous sommes iniialement partis du principe que nous avions besoin de sélectioner des gènes parmi nos 3117 gènes initiaux.</p>

<p>On regarde tout d'abord la corrélation entre les gènes. On plot le nombre de gènes selon le coefficient de corrélation.</p>

<div class="highlight highlight-source-r"><pre><span class="pl-en">geneint</span><span class="pl-k">&lt;-</span> <span class="pl-k">function</span>(<span class="pl-smi">seuil</span>){<span class="pl-smi">gene</span>[,abs(cor(<span class="pl-smi">gene</span>[,<span class="pl-c1">1</span><span class="pl-k">:</span>ncol(<span class="pl-smi">gene</span>)], <span class="pl-smi">cholesterol</span>))<span class="pl-k">&gt;</span><span class="pl-smi">seuil</span>]}

<span class="pl-v">visual</span> <span class="pl-k">=</span> c()
<span class="pl-k">for</span> ( <span class="pl-smi">i</span> <span class="pl-k">in</span> seq(<span class="pl-c1">0</span>, <span class="pl-c1">1</span> , <span class="pl-c1">0.1</span>))
  {
    <span class="pl-v">visual</span> <span class="pl-k">=</span> c(<span class="pl-smi">visual</span>, ncol(geneint(<span class="pl-smi">i</span>)))
}
plot(seq(<span class="pl-c1">0</span>, <span class="pl-c1">1</span> , <span class="pl-c1">0.1</span>), <span class="pl-smi">visual</span>)

<span class="pl-smi">workgene</span> <span class="pl-k">&lt;-</span> geneint(<span class="pl-c1">0.4</span>)

image(abs(cor(<span class="pl-smi">workgene</span>))) 
</pre></div>

<p>En faisant la heatmap des gènes présentant une forte corrélation avec le cholestérol (&gt;0.4) (environ 500 gènes d'intérêt), nous pouvons constater que nous avons un grand nombre de gènes corrélés entre eux. Ce n'est pas un bon descripteur pour nos données.</p>

<p>Nous tentons ensuite une sélection sur la variance.</p>

<div class="highlight highlight-source-r"><pre>library(<span class="pl-smi">leaps</span>)

<span class="pl-smi">var.gene</span><span class="pl-k">&lt;-</span>apply(<span class="pl-smi">gene</span>, <span class="pl-c1">2</span>, <span class="pl-smi">var</span>)

plot(<span class="pl-smi">var.gene</span>)
hist(<span class="pl-smi">var.gene</span>, <span class="pl-v">breaks</span><span class="pl-k">=</span><span class="pl-c1">100</span>)

<span class="pl-en">plot.gene.var</span><span class="pl-k">&lt;-</span><span class="pl-k">function</span>(<span class="pl-smi">min</span>)
{
  <span class="pl-smi">step</span><span class="pl-k">&lt;-</span>seq(<span class="pl-smi">min</span>, <span class="pl-c1">0.5</span>, <span class="pl-c1">0.01</span>)
  <span class="pl-smi">n.genes</span><span class="pl-k">&lt;-</span>c()
  <span class="pl-k">for</span>(<span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-smi">step</span>)
  {
    <span class="pl-smi">n.genes</span><span class="pl-k">&lt;-</span>c(<span class="pl-smi">n.genes</span>, sum(<span class="pl-smi">var.gene</span><span class="pl-k">&gt;</span><span class="pl-smi">i</span>))
  }
  plot(<span class="pl-smi">step</span>, <span class="pl-smi">n.genes</span>)
}
</pre></div>

<div class="highlight highlight-source-r"><pre>par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>c(<span class="pl-c1">2</span>,<span class="pl-c1">2</span>))
plot.gene.var(<span class="pl-c1">0</span>)
plot.gene.var(<span class="pl-c1">0.04</span>)
plot.gene.var(<span class="pl-c1">0.08</span>)
plot.gene.var(<span class="pl-c1">0.2</span>)

<span class="pl-smi">work.gene</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">gene</span>[,<span class="pl-smi">var.gene</span><span class="pl-k">&gt;</span><span class="pl-c1">0.1</span>]
ncol(<span class="pl-smi">work.gene</span>)
</pre></div>

<p>En regardant la distribution de la variance des 3117 gènes, la majorité est proche de 0 avec seulement 49 présentant une variance supérieure à 0.1.</p>

<p>\newpage</p>

<h2><a id="user-content-régression-stepwise" class="anchor" href="#régression-stepwise" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Régression Stepwise</h2>

<p>Nous avons fait une régression stepwise sur les gènes sélectionnés sur la variance.</p>

<div class="highlight highlight-source-r"><pre><span class="pl-smi">scope</span> <span class="pl-k">&lt;-</span> <span class="pl-k">list</span>(<span class="pl-v">lower</span> <span class="pl-k">=</span> terms(<span class="pl-smi">cholesterol</span> <span class="pl-k">~</span> <span class="pl-c1">1</span>, <span class="pl-v">data</span><span class="pl-k">=</span><span class="pl-smi">work.gene</span>), <span class="pl-v">upper</span> <span class="pl-k">=</span> terms(<span class="pl-smi">cholesterol</span> <span class="pl-k">~</span> ., <span class="pl-v">data</span><span class="pl-k">=</span><span class="pl-smi">work.gene</span>))
<span class="pl-smi">step.AIC</span> <span class="pl-k">&lt;-</span> step( lm(<span class="pl-smi">cholesterol</span><span class="pl-k">~</span><span class="pl-c1">1</span>, <span class="pl-smi">work.gene</span>), <span class="pl-smi">scope</span>, <span class="pl-v">direction</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>both<span class="pl-pds">'</span></span>, <span class="pl-v">trace</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)
<span class="pl-smi">step.BIC</span> <span class="pl-k">&lt;-</span> step( lm(<span class="pl-smi">cholesterol</span><span class="pl-k">~</span><span class="pl-c1">1</span>, <span class="pl-smi">work.gene</span>), <span class="pl-smi">scope</span>, <span class="pl-v">direction</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">'</span>both<span class="pl-pds">'</span></span>, <span class="pl-v">k</span><span class="pl-k">=</span>log(nrow(<span class="pl-smi">work.gene</span>)), <span class="pl-v">trace</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)

<span class="pl-smi">step.AIC</span>

<span class="pl-smi">step.BIC</span>

par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>c(<span class="pl-c1">2</span>,<span class="pl-c1">2</span>))
plot(<span class="pl-smi">step.BIC</span>, <span class="pl-v">which</span><span class="pl-k">=</span>c(<span class="pl-c1">1</span>,<span class="pl-c1">2</span>,<span class="pl-c1">4</span>,<span class="pl-c1">5</span>))</pre></div>

<p>\newpage</p>

<h2><a id="user-content-régression-ridge" class="anchor" href="#régression-ridge" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Régression Ridge</h2>

<p>Nous avons fait une régression stepwise sur les gènes sélectionnés sur la variance.</p>

<div class="highlight highlight-source-r"><pre>library(<span class="pl-smi">glmnet</span>)
<span class="pl-smi">ridge</span><span class="pl-k">&lt;-</span>glmnet(as.matrix(<span class="pl-smi">work.gene</span>), <span class="pl-smi">cholesterol</span>, <span class="pl-v">alpha</span><span class="pl-k">=</span><span class="pl-c1">0</span>)
par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>c(<span class="pl-c1">1</span>,<span class="pl-c1">3</span>))
plot(<span class="pl-smi">ridge</span>, <span class="pl-v">xvar</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>lambda<span class="pl-pds">"</span></span>)
plot(<span class="pl-smi">ridge</span>, <span class="pl-v">xvar</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>norm<span class="pl-pds">"</span></span>)
plot(<span class="pl-smi">ridge</span>, <span class="pl-v">xvar</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>dev<span class="pl-pds">"</span></span>)

<span class="pl-smi">ridge.10cv</span> <span class="pl-k">&lt;-</span> cv.glmnet(as.matrix(<span class="pl-smi">work.gene</span>),<span class="pl-smi">cholesterol</span>,<span class="pl-v">nfolds</span><span class="pl-k">=</span><span class="pl-c1">10</span>, <span class="pl-v">alpha</span><span class="pl-k">=</span><span class="pl-c1">0</span>, <span class="pl-v">grouped</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)
<span class="pl-smi">ridge.loo</span> <span class="pl-k">&lt;-</span> cv.glmnet(as.matrix(<span class="pl-smi">work.gene</span>),<span class="pl-smi">cholesterol</span>,<span class="pl-v">nfolds</span><span class="pl-k">=</span>nrow(<span class="pl-smi">work.gene</span>), <span class="pl-v">alpha</span><span class="pl-k">=</span><span class="pl-c1">0</span> , <span class="pl-v">grouped</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)
par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>c(<span class="pl-c1">1</span>,<span class="pl-c1">2</span>))
plot(<span class="pl-smi">ridge.10cv</span>)
plot(<span class="pl-smi">ridge.loo</span>)

cbind(predict(<span class="pl-smi">ridge</span>, as.matrix(<span class="pl-smi">work.gene</span>[<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">10</span>,]), <span class="pl-v">s</span><span class="pl-k">=</span><span class="pl-smi">ridge.10cv</span><span class="pl-k">$</span><span class="pl-smi">lambda.min</span>), <span class="pl-smi">cholesterol</span>[<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">10</span>])

cbind(predict(<span class="pl-smi">ridge</span>, as.matrix(<span class="pl-smi">work.gene</span>[<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">10</span>,]), <span class="pl-v">s</span><span class="pl-k">=</span><span class="pl-smi">ridge.10cv</span><span class="pl-k">$</span><span class="pl-smi">lambda.1se</span> ), <span class="pl-smi">cholesterol</span>[<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">10</span>])</pre></div>

<p>On voit pour les gènes X211 et X213, que selon le modèle réalisé, les gènes ont une bonne moyenne mais une variance trop faible.</p>

<p>\newpage</p>

<h2><a id="user-content-régression-lasso" class="anchor" href="#régression-lasso" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Régression Lasso</h2>

<p>Nous avons fait une régression lasso sur les gènes sélectionnés sur la variance.</p>

<div class="highlight highlight-source-r"><pre><span class="pl-smi">lasso</span> <span class="pl-k">&lt;-</span> glmnet(as.matrix(<span class="pl-smi">work.gene</span>), <span class="pl-smi">cholesterol</span>)
par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>c(<span class="pl-c1">1</span>,<span class="pl-c1">3</span>))
plot(<span class="pl-smi">lasso</span>, <span class="pl-v">xvar</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>lambda<span class="pl-pds">"</span></span>)
plot(<span class="pl-smi">lasso</span>, <span class="pl-v">xvar</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>norm<span class="pl-pds">"</span></span>)
plot(<span class="pl-smi">lasso</span>, <span class="pl-v">xvar</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>dev<span class="pl-pds">"</span></span>)

<span class="pl-smi">lasso.10cv</span> <span class="pl-k">&lt;-</span> cv.glmnet(as.matrix(<span class="pl-smi">work.gene</span>),<span class="pl-smi">cholesterol</span>,<span class="pl-v">nfolds</span><span class="pl-k">=</span><span class="pl-c1">10</span>, <span class="pl-v">grouped</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)
<span class="pl-smi">lasso.loo</span> <span class="pl-k">&lt;-</span> cv.glmnet(as.matrix(<span class="pl-smi">work.gene</span>),<span class="pl-smi">cholesterol</span>,<span class="pl-v">nfolds</span><span class="pl-k">=</span>nrow(<span class="pl-smi">work.gene</span>) , <span class="pl-v">grouped</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)
par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>c(<span class="pl-c1">1</span>,<span class="pl-c1">2</span>))
plot(<span class="pl-smi">lasso.10cv</span>)
plot(<span class="pl-smi">lasso.loo</span>)

cbind(predict(<span class="pl-smi">lasso</span>, as.matrix(<span class="pl-smi">work.gene</span>[<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">10</span>,]), <span class="pl-v">s</span><span class="pl-k">=</span><span class="pl-smi">lasso.10cv</span><span class="pl-k">$</span><span class="pl-smi">lambda.min</span>), <span class="pl-smi">cholesterol</span>[<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">10</span>])

cbind(predict(<span class="pl-smi">lasso</span>, as.matrix(<span class="pl-smi">work.gene</span>[<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">10</span>,]), <span class="pl-v">s</span><span class="pl-k">=</span><span class="pl-smi">lasso.10cv</span><span class="pl-k">$</span><span class="pl-smi">lambda.1se</span>), <span class="pl-smi">cholesterol</span>[<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">10</span>])</pre></div>

<p>Là encore, les gènes ont une bonne moyenne mais une variance trop faible.</p>

<div class="highlight highlight-source-r"><pre><span class="pl-smi">n</span> <span class="pl-k">&lt;-</span> nrow(<span class="pl-smi">work.gene</span>)
<span class="pl-smi">p</span> <span class="pl-k">&lt;-</span> ncol(<span class="pl-smi">work.gene</span>) <span class="pl-k">-</span> <span class="pl-c1">1</span> <span class="pl-k">+</span> <span class="pl-c1">1</span>
<span class="pl-smi">AIC</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">n</span><span class="pl-k">*</span>log(colMeans((<span class="pl-smi">cholesterol</span> <span class="pl-k">-</span> predict(<span class="pl-smi">lasso</span>, as.matrix(<span class="pl-smi">work.gene</span>)))<span class="pl-k">^</span><span class="pl-c1">2</span>)) <span class="pl-k">+</span> <span class="pl-c1">2</span> <span class="pl-k">*</span> <span class="pl-smi">lasso</span><span class="pl-k">$</span><span class="pl-smi">df</span>
<span class="pl-smi">BIC</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">n</span><span class="pl-k">*</span>log(colMeans((<span class="pl-smi">cholesterol</span> <span class="pl-k">-</span> predict(<span class="pl-smi">lasso</span>, as.matrix(<span class="pl-smi">work.gene</span>)))<span class="pl-k">^</span><span class="pl-c1">2</span>)) <span class="pl-k">+</span> log(<span class="pl-smi">n</span>) <span class="pl-k">*</span> <span class="pl-smi">lasso</span><span class="pl-k">$</span><span class="pl-smi">df</span>
<span class="pl-smi">eBIC</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">n</span><span class="pl-k">*</span>log(colMeans((<span class="pl-smi">cholesterol</span> <span class="pl-k">-</span> predict(<span class="pl-smi">lasso</span>, as.matrix(<span class="pl-smi">work.gene</span>)))<span class="pl-k">^</span><span class="pl-c1">2</span>)) <span class="pl-k">+</span> log(<span class="pl-smi">p</span>) <span class="pl-k">*</span> <span class="pl-smi">lasso</span><span class="pl-k">$</span><span class="pl-smi">df</span>
<span class="pl-smi">mBIC</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">n</span><span class="pl-k">*</span>log(colMeans((<span class="pl-smi">cholesterol</span> <span class="pl-k">-</span> predict(<span class="pl-smi">lasso</span>, as.matrix(<span class="pl-smi">work.gene</span>)))<span class="pl-k">^</span><span class="pl-c1">2</span>)) <span class="pl-k">+</span> (log(<span class="pl-smi">n</span>) <span class="pl-k">+</span> <span class="pl-c1">2</span> <span class="pl-k">*</span>log(<span class="pl-smi">p</span>)) <span class="pl-k">*</span> <span class="pl-smi">lasso</span><span class="pl-k">$</span><span class="pl-smi">df</span>

library(<span class="pl-smi">ggplot2</span>)
</pre></div>

<p>Ecart à la moyenne du modèle pour chacune des valeurs lambda :</p>

<div class="highlight highlight-source-r"><pre><span class="pl-smi">d</span> <span class="pl-k">&lt;-</span> <span class="pl-k">data.frame</span>(<span class="pl-v">lambda</span> <span class="pl-k">=</span> rep(<span class="pl-smi">lasso</span><span class="pl-k">$</span><span class="pl-smi">lambda</span>, <span class="pl-c1">4</span>),
<span class="pl-v">value</span> <span class="pl-k">=</span> c(<span class="pl-smi">AIC</span>, <span class="pl-smi">BIC</span>, <span class="pl-smi">eBIC</span>, <span class="pl-smi">mBIC</span>),
<span class="pl-v">critere</span> <span class="pl-k">=</span> <span class="pl-k">factor</span>(rep(c(<span class="pl-s"><span class="pl-pds">"</span>AIC<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>BIC<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>eBIC<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>mBIC<span class="pl-pds">"</span></span>), <span class="pl-v">each</span><span class="pl-k">=</span>length(<span class="pl-smi">lasso</span><span class="pl-k">$</span><span class="pl-smi">lambda</span>))))
ggplot(<span class="pl-smi">d</span>, aes(<span class="pl-v">x</span><span class="pl-k">=</span><span class="pl-smi">lambda</span>,<span class="pl-v">y</span><span class="pl-k">=</span><span class="pl-smi">value</span>,<span class="pl-v">colour</span><span class="pl-k">=</span><span class="pl-smi">critere</span>,<span class="pl-v">group</span><span class="pl-k">=</span><span class="pl-smi">critere</span>)) <span class="pl-k">+</span> geom_line() <span class="pl-k">+</span> scale_x_log10()

<span class="pl-smi">lambda.min.BIC</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">lasso</span><span class="pl-k">$</span><span class="pl-smi">lambda</span>[which.min(<span class="pl-smi">BIC</span>)]
<span class="pl-smi">lambda.min.mBIC</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">lasso</span><span class="pl-k">$</span><span class="pl-smi">lambda</span>[which.min(<span class="pl-smi">mBIC</span>)]

cbind(predict(<span class="pl-smi">lasso</span>, as.matrix(<span class="pl-smi">work.gene</span>[<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">10</span>,]), <span class="pl-v">s</span><span class="pl-k">=</span><span class="pl-smi">lambda.min.BIC</span>), <span class="pl-smi">cholesterol</span>[<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">10</span>])

cbind(predict(<span class="pl-smi">lasso</span>, as.matrix(<span class="pl-smi">work.gene</span>[<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">10</span>,]), <span class="pl-v">s</span><span class="pl-k">=</span><span class="pl-smi">lambda.min.mBIC</span>), <span class="pl-smi">cholesterol</span>[<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">10</span>])
</pre></div>

<p>Les modèles trouvés ont une bonne espérance mais une variance quasiment nulle, ne correspondant pas à la vraie variance des données de Cholesterol.</p>

<p>\newpage</p>

<h2><a id="user-content-régression-sur-lensemble-des-gènes" class="anchor" href="#régression-sur-lensemble-des-gènes" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Régression sur l'ensemble des gènes</h2>

<p>Suite aux résultats du lasso et du ridge, nous voulons faire une régression sur l'ensemble des gènes.
Nous allons donc faire une méthode Elastic net qui, avec un paramètre alpha, permet de combiner les deux techniques. Alpha = 0 : ridge penalty,  Alpha = 1 : lasso penaly</p>

<p>Nous cherchons donc l'alpha qui minimise la différence de variance entre les données prédites et les vraies données en faisant également une validation croisée.</p>

<div class="highlight highlight-source-r"><pre><span class="pl-smi">var.chol</span><span class="pl-k">&lt;-</span>var(<span class="pl-smi">cholesterol</span>)

<span class="pl-smi">better.model.loo</span> <span class="pl-k">&lt;-</span> cv.glmnet(as.matrix(<span class="pl-smi">gene</span>),<span class="pl-smi">cholesterol</span>,<span class="pl-v">nfolds</span><span class="pl-k">=</span>nrow(<span class="pl-smi">gene</span>) , <span class="pl-v">grouped</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)

<span class="pl-en">err.model</span><span class="pl-k">&lt;-</span><span class="pl-k">function</span>(<span class="pl-smi">a</span>){
  <span class="pl-smi">model</span><span class="pl-k">&lt;-</span>glmnet(as.matrix(<span class="pl-smi">gene</span>), <span class="pl-smi">cholesterol</span>, <span class="pl-v">alpha</span><span class="pl-k">=</span><span class="pl-smi">a</span>)
  <span class="pl-k">return</span>(var(predict(<span class="pl-smi">model</span>, as.matrix(<span class="pl-smi">gene</span>), <span class="pl-v">s</span><span class="pl-k">=</span><span class="pl-smi">better.model.loo</span><span class="pl-k">$</span><span class="pl-smi">lambda.min</span>))<span class="pl-k">-</span><span class="pl-smi">var.chol</span>)
}

<span class="pl-smi">errors</span><span class="pl-k">&lt;-</span>lapply(seq(<span class="pl-c1">0</span>,<span class="pl-c1">1</span>,<span class="pl-c1">0.1</span>), <span class="pl-smi">err.model</span>)

plot(seq(<span class="pl-c1">0</span>,<span class="pl-c1">1</span>,<span class="pl-c1">0.1</span>), <span class="pl-smi">errors</span>)

<span class="pl-smi">errors</span><span class="pl-k">&lt;-</span>lapply(seq(<span class="pl-c1">0</span>,<span class="pl-c1">0.1</span>,<span class="pl-c1">0.01</span>), <span class="pl-smi">err.model</span>)

plot(seq(<span class="pl-c1">0</span>,<span class="pl-c1">0.1</span>,<span class="pl-c1">0.01</span>), <span class="pl-smi">errors</span>)

<span class="pl-smi">better.model</span><span class="pl-k">&lt;-</span>glmnet(as.matrix(<span class="pl-smi">gene</span>), <span class="pl-smi">cholesterol</span>, <span class="pl-v">alpha</span><span class="pl-k">=</span><span class="pl-c1">0.04</span>)

<span class="pl-smi">better.model.lambda.min</span><span class="pl-k">&lt;-</span>cv.glmnet(as.matrix(<span class="pl-smi">gene</span>),<span class="pl-smi">cholesterol</span>,<span class="pl-v">nfolds</span><span class="pl-k">=</span>nrow(<span class="pl-smi">gene</span>) , <span class="pl-v">grouped</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)<span class="pl-k">$</span><span class="pl-smi">lambda.min</span>

<span class="pl-smi">err.var.min</span><span class="pl-k">&lt;-</span><span class="pl-smi">var.chol</span><span class="pl-k">-</span>var(predict(<span class="pl-smi">better.model</span>, as.matrix(<span class="pl-smi">gene</span>), <span class="pl-v">s</span><span class="pl-k">=</span><span class="pl-smi">better.model.lambda.min</span>))

<span class="pl-smi">err.mean.min</span><span class="pl-k">&lt;-</span>mean(<span class="pl-smi">cholesterol</span>)<span class="pl-k">-</span>mean(predict(<span class="pl-smi">better.model</span>, as.matrix(<span class="pl-smi">gene</span>), <span class="pl-v">s</span><span class="pl-k">=</span><span class="pl-smi">better.model.lambda.min</span>))

<span class="pl-smi">better.model.lambda.1se</span><span class="pl-k">&lt;-</span>cv.glmnet(as.matrix(<span class="pl-smi">gene</span>),<span class="pl-smi">cholesterol</span>,<span class="pl-v">nfolds</span><span class="pl-k">=</span>nrow(<span class="pl-smi">gene</span>) , <span class="pl-v">grouped</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)<span class="pl-k">$</span><span class="pl-smi">lambda.1se</span>

<span class="pl-smi">err.var.1se</span><span class="pl-k">&lt;-</span><span class="pl-smi">var.chol</span><span class="pl-k">-</span>var(predict(<span class="pl-smi">better.model</span>, as.matrix(<span class="pl-smi">gene</span>), <span class="pl-v">s</span><span class="pl-k">=</span><span class="pl-smi">better.model.lambda.1se</span>))

<span class="pl-smi">err.mean.1se</span><span class="pl-k">&lt;-</span>mean(<span class="pl-smi">cholesterol</span>)<span class="pl-k">-</span>mean(predict(<span class="pl-smi">better.model</span>, as.matrix(<span class="pl-smi">gene</span>), <span class="pl-v">s</span><span class="pl-k">=</span><span class="pl-smi">better.model.lambda.1se</span>))

<span class="pl-k">matrix</span>(c(<span class="pl-smi">err.var.min</span>, <span class="pl-smi">err.var.1se</span>, <span class="pl-smi">err.mean.min</span>, <span class="pl-smi">err.mean.1se</span>), <span class="pl-c1">2</span>, <span class="pl-c1">2</span>, <span class="pl-v">dimnames</span><span class="pl-k">=</span><span class="pl-k">list</span>(c(<span class="pl-s"><span class="pl-pds">"</span>lambda.min<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>lambda.1se<span class="pl-pds">"</span></span>), c(<span class="pl-s"><span class="pl-pds">"</span>err.var<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>err.mean<span class="pl-pds">"</span></span>)))</pre></div>

<p>En tracant le plot des données de cholesterol observées et des données prédites avec un elastic net (alpha = 0.04), nous pouvons observer que le modèle comporte 346 gènes et semble très satisfaisant.</p>

<div class="highlight highlight-source-r"><pre>par(<span class="pl-v">mfrow</span><span class="pl-k">=</span>c(<span class="pl-c1">1</span>,<span class="pl-c1">1</span>))

<span class="pl-smi">predicted.cholesterol</span><span class="pl-k">&lt;-</span>predict(<span class="pl-smi">better.model</span>, as.matrix(<span class="pl-smi">gene</span>), <span class="pl-v">s</span><span class="pl-k">=</span><span class="pl-smi">better.model.loo</span><span class="pl-k">$</span><span class="pl-smi">lambda.min</span>)
plot(<span class="pl-smi">predicted.cholesterol</span>, <span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>)
legend(<span class="pl-s"><span class="pl-pds">"</span>bottomleft<span class="pl-pds">"</span></span>, <span class="pl-v">legend</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>Données prédites<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Données observées<span class="pl-pds">"</span></span>), <span class="pl-v">pch</span> <span class="pl-k">=</span> <span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">1</span>, <span class="pl-v">col</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>green<span class="pl-pds">"</span></span>))
points(<span class="pl-smi">cholesterol</span>, <span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>green<span class="pl-pds">"</span></span>)
</pre></div>

<p>\newpage</p>

<h2><a id="user-content-facteurs-du-modèle" class="anchor" href="#facteurs-du-modèle" aria-hidden="true"><svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Facteurs du modèle</h2>

<p>On remarque que les 346 gènes semblent être organisés en au moins 2 groupes</p>

<div class="highlight highlight-source-r"><pre><span class="pl-smi">better.model.coef</span><span class="pl-k">&lt;-</span>predict(<span class="pl-smi">better.model</span>, <span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>coef<span class="pl-pds">"</span></span>, <span class="pl-v">s</span><span class="pl-k">=</span><span class="pl-smi">better.model.loo</span><span class="pl-k">$</span><span class="pl-smi">lambda.min</span>)

sum(<span class="pl-smi">better.model.coef</span><span class="pl-k">!=</span><span class="pl-c1">0</span>)

<span class="pl-smi">coef.names</span><span class="pl-k">&lt;-</span>c(dimnames(<span class="pl-smi">better.model.coef</span>)[[<span class="pl-c1">1</span>]][<span class="pl-c1">1</span>], dimnames(<span class="pl-smi">better.model.coef</span>)[[<span class="pl-c1">1</span>]][<span class="pl-smi">better.model.coef</span><span class="pl-k">@</span><span class="pl-smi">i</span>])
<span class="pl-smi">better.model.coef</span><span class="pl-k">&lt;-</span><span class="pl-smi">better.model.coef</span><span class="pl-k">@</span><span class="pl-smi">x</span>[<span class="pl-smi">better.model.coef</span><span class="pl-k">@</span><span class="pl-smi">x</span><span class="pl-k">!=</span><span class="pl-c1">0</span>]
names(<span class="pl-smi">better.model.coef</span>)<span class="pl-k">&lt;-</span><span class="pl-smi">coef.names</span>

head(<span class="pl-smi">better.model.coef</span>)

<span class="pl-smi">gene.dist</span><span class="pl-k">&lt;-</span>dist(<span class="pl-smi">gene</span>[<span class="pl-smi">coef.names</span>[<span class="pl-k">-</span><span class="pl-c1">1</span>]])
plot(hclust(<span class="pl-smi">gene.dist</span>))
</pre></div>
</article>
  </div>

</div>

<button type="button" data-facebox="#jump-to-line" data-facebox-class="linejump" data-hotkey="l" class="d-none">Jump to Line</button>
<div id="jump-to-line" style="display:none">
  <!-- '"` --><!-- </textarea></xmp> --></option></form><form accept-charset="UTF-8" action="" class="js-jump-to-line-form" method="get"><div style="margin:0;padding:0;display:inline"><input name="utf8" type="hidden" value="&#x2713;" /></div>
    <input class="form-control linejump-input js-jump-to-line-field" type="text" placeholder="Jump to line&hellip;" aria-label="Jump to line" autofocus>
    <button type="submit" class="btn">Go</button>
</form></div>

  </div>
  <div class="modal-backdrop js-touch-events"></div>
</div>


    </div>
  </div>

    </div>

        <div class="container site-footer-container">
  <div class="site-footer" role="contentinfo">
    <ul class="site-footer-links float-right">
        <li><a href="https://github.com/contact" data-ga-click="Footer, go to contact, text:contact">Contact GitHub</a></li>
      <li><a href="https://developer.github.com" data-ga-click="Footer, go to api, text:api">API</a></li>
      <li><a href="https://training.github.com" data-ga-click="Footer, go to training, text:training">Training</a></li>
      <li><a href="https://shop.github.com" data-ga-click="Footer, go to shop, text:shop">Shop</a></li>
        <li><a href="https://github.com/blog" data-ga-click="Footer, go to blog, text:blog">Blog</a></li>
        <li><a href="https://github.com/about" data-ga-click="Footer, go to about, text:about">About</a></li>

    </ul>

    <a href="https://github.com" aria-label="Homepage" class="site-footer-mark" title="GitHub">
      <svg aria-hidden="true" class="octicon octicon-mark-github" height="24" version="1.1" viewBox="0 0 16 16" width="24"><path fill-rule="evenodd" d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0 0 16 8c0-4.42-3.58-8-8-8z"/></svg>
</a>
    <ul class="site-footer-links">
      <li>&copy; 2016 <span title="0.04606s from github-fe-b03c132.cp1-iad.github.net">GitHub</span>, Inc.</li>
        <li><a href="https://github.com/site/terms" data-ga-click="Footer, go to terms, text:terms">Terms</a></li>
        <li><a href="https://github.com/site/privacy" data-ga-click="Footer, go to privacy, text:privacy">Privacy</a></li>
        <li><a href="https://github.com/security" data-ga-click="Footer, go to security, text:security">Security</a></li>
        <li><a href="https://status.github.com/" data-ga-click="Footer, go to status, text:status">Status</a></li>
        <li><a href="https://help.github.com" data-ga-click="Footer, go to help, text:help">Help</a></li>
    </ul>
  </div>
</div>



    

    <div id="ajax-error-message" class="ajax-error-message flash flash-error">
      <svg aria-hidden="true" class="octicon octicon-alert" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M8.865 1.52c-.18-.31-.51-.5-.87-.5s-.69.19-.87.5L.275 13.5c-.18.31-.18.69 0 1 .19.31.52.5.87.5h13.7c.36 0 .69-.19.86-.5.17-.31.18-.69.01-1L8.865 1.52zM8.995 13h-2v-2h2v2zm0-3h-2V6h2v4z"/></svg>
      <button type="button" class="flash-close js-flash-close js-ajax-error-dismiss" aria-label="Dismiss error">
        <svg aria-hidden="true" class="octicon octicon-x" height="16" version="1.1" viewBox="0 0 12 16" width="12"><path fill-rule="evenodd" d="M7.48 8l3.75 3.75-1.48 1.48L6 9.48l-3.75 3.75-1.48-1.48L4.52 8 .77 4.25l1.48-1.48L6 6.52l3.75-3.75 1.48 1.48z"/></svg>
      </button>
      You can't perform that action at this time.
    </div>


      
      <script crossorigin="anonymous" integrity="sha256-Nx6NFlbsuXNwgxytKPfrySbpY2rv9pnnWl6tPk7YVhg=" src="https://assets-cdn.github.com/assets/frameworks-371e8d1656ecb97370831cad28f7ebc926e9636aeff699e75a5ead3e4ed85618.js"></script>
      <script async="async" crossorigin="anonymous" integrity="sha256-q1O55iRgoevO9Kq555Y7W9JJdHRoaWTSfkznVqZfp+A=" src="https://assets-cdn.github.com/assets/github-ab53b9e62460a1ebcef4aab9e7963b5bd2497474686964d27e4ce756a65fa7e0.js"></script>
      
      
      
      
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-banner d-none">
      <svg aria-hidden="true" class="octicon octicon-alert" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M8.865 1.52c-.18-.31-.51-.5-.87-.5s-.69.19-.87.5L.275 13.5c-.18.31-.18.69 0 1 .19.31.52.5.87.5h13.7c.36 0 .69-.19.86-.5.17-.31.18-.69.01-1L8.865 1.52zM8.995 13h-2v-2h2v2zm0-3h-2V6h2v4z"/></svg>
      <span class="signed-in-tab-flash">You signed in with another tab or window. <a href="">Reload</a> to refresh your session.</span>
      <span class="signed-out-tab-flash">You signed out in another tab or window. <a href="">Reload</a> to refresh your session.</span>
    </div>
    <div class="facebox" id="facebox" style="display:none;">
  <div class="facebox-popup">
    <div class="facebox-content" role="dialog" aria-labelledby="facebox-header" aria-describedby="facebox-description">
    </div>
    <button type="button" class="facebox-close js-facebox-close" aria-label="Close modal">
      <svg aria-hidden="true" class="octicon octicon-x" height="16" version="1.1" viewBox="0 0 12 16" width="12"><path fill-rule="evenodd" d="M7.48 8l3.75 3.75-1.48 1.48L6 9.48l-3.75 3.75-1.48-1.48L4.52 8 .77 4.25l1.48-1.48L6 6.52l3.75-3.75 1.48 1.48z"/></svg>
    </button>
  </div>
</div>

  </body>
</html>

